commit d2e85432a2e0a6f31bd9489800f443228f020ed6
Author: Yang Yingliang <yangyingliang@huawei.com>
Date:   Fri Aug 26 15:33:37 2022 +0800

    fail_function: fix wrong use of fei_attr_remove()
    
    If register_kprobe() fails, the new attr is not added to the list yet, so
    it should call fei_attr_free() intstead.
    
    Link: https://lkml.kernel.org/r/20220826073337.2085798-3-yangyingliang@huawei.com
    Fixes: 4b1a29a7f542 ("error-injection: Support fault injection framework")
    Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
    Reviewed-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>

diff --git a/kernel/fail_function.c b/kernel/fail_function.c
index 893e8f9a9118..a7ccd2930c5f 100644
--- a/kernel/fail_function.c
+++ b/kernel/fail_function.c
@@ -295,7 +295,7 @@ static ssize_t fei_write(struct file *file, const char __user *buffer,
 
 	ret = register_kprobe(&attr->kp);
 	if (ret) {
-		fei_attr_remove(attr);
+		fei_attr_free(attr);
 		goto out;
 	}
 	fei_debugfs_add_attr(attr);